iXBRLViewerPlugin/iXBRLViewer.py

```diff
@@ -546,7 +546,7 @@ def addFile(self, ivf):
     def addFilingDoc(self, filingDocuments):
         self.filingDocuments = filingDocuments
 
-    def save(self, destination: io.BytesIO | str, zipOutput: bool = False, copyScriptPath: Path | None = None):
+    def save(self, destination: io.BytesIO | str, zipOutput: bool = False, copyScriptPath: Path | None = None, stubFileOnly: bool = False):
```

Comment:
I think we should probably catch `stubFileOnly` used without `useStubViewer` and throw an error, as currently you won't get an error, you'll just silently get the first file from the document set.

Perhaps we could move the option on to `createViewer` and use it to control the calls to `addFile`? This might be clearer than just only outputting the first file.